### PR TITLE
Feature/add kucoin adapter

### PR DESCRIPTION
**Pull request by philbow61**
## Description

added the kucoin adapter 

## Other changes

refactored one bitstamp adapter test

## Tested

- tested each of the functions in the adapter
- ran oracle locally

## Related issues

- Fixes https://github.com/mento-protocol/mento-general/issues/121

### CODE REVIEW

File: test/exchange_adapters/kucoin.test.ts [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+  describe('isOrderbookLive', () => {
+    it('returns true if enableTrading is true', async () => {
+      jest
+        .spyOn(kucoinAdapter, 'fetchFromApi')
+        .mockReturnValue(Promise.resolve(validMockStatusJson))
+      expect(await kucoinAdapter.isOrderbookLive()).toEqual(true)
+    })
+
+    it('returns false if enableTrading is false', async () => {
```
Comment (philbow61) [inline test/exchange_adapters/kucoin.test.ts:239]: https://github.com/celo-org/celo-oracle/pull/145/files#r1085219901

File: test/exchange_adapters/kucoin.test.ts [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+        isMarginEnabled: true,
+        enableTrading: true,
+      },
+    ],
+  }
+  const inValidMockStatusJson = {
+    code: '200000',
+    data: [
+      {
+        symbol: 'LOKI-BTC',
```
Comment (tobikuhlmann) [inline test/exchange_adapters/kucoin.test.ts:191]: why a second pair here for order book status check?

**Review by tobikuhlmann — APPROVED** [review]
- [x] tests look good and pass
- [x] runs locally and reports correct KuCoin pair prices

LGTM

File: test/exchange_adapters/kucoin.test.ts (hunk 2 above)
Comment (philbow61) [reply test/exchange_adapters/kucoin.test.ts:191]: To ensure that the book status function parses the correct pair.
The API response returns information for all pairs

Comment (tobikuhlmann) [reply test/exchange_adapters/kucoin.test.ts:191]: alright 👍

**Pull request merged**